src/store/storage-controller.ts

```diff
@@ -33,9 +35,9 @@ export class StorageController {
     // but NOTE: it is possible that a data stream is not given in such case, for instance,
     // a subsequent RecordsWrite that changes the `published` property, but the data hasn't changed,
     // in this case requiring re-uploading of the data is extremely inefficient so we take care allow omission of data stream
-    if (message.descriptor.dataCid !== undefined) {
-      const messageCid = await Message.getCid(message);
+    const messageCid = await Message.getCid(message);
```

Comment:
NIT: this should probably go before the comment, as the comment pertains to the `if` 😅